### ISSUE TITLE
Design issue: Always have buttons floating bottom right.

### ISSUE THREAD

**Issue by cdrage** [containers/podman-desktop#5926]
### Is your enhancement related to a problem? Please describe

My muscle memory always thinks that the buttons are going to be on the bottom right similar to onboarding:

![image](https://github.com/containers/podman-desktop/assets/6422176/87b5c0bf-6f27-4f5d-84de-ce0a0ad8bba5)

and see issue: https://github.com/containers/podman-desktop/issues/5164

Some of the other sections it's centered in the form, as well as you have to scroll down to it or there is too much white space:

![image](https://github.com/containers/podman-desktop/assets/6422176/86490990-aefb-470a-9ff5-4a5d1b2e191b)

Other sections also include resources which is also different:

![image](https://github.com/containers/podman-desktop/assets/6422176/38b6a7a7-bd3c-488c-9530-14a79c07bceb)

### Describe the solution you'd like

I propose similar to onboarding, we "float" any action buttons on the bottom for *all* sections of Podman Desktop  (with the exception of dialogs) similar to the current onboarding buttons so it's more easily accessible (no need to scroll) as well as good for muscle memory. 

### Additional context

@mairin

**Comment by mairin** [comment]
@ekidneyrh Do you have cycles to explore the visual design for this in the Form screen flex mockups you created?

**Comment by ekidneyrh** [comment]
Hey! The screenshots you added have either the cancel button in a box vs having it as text beside the actionable button. 

I think having the cancel button in a box with a white stoke would keep the consistency with the modal dialogue that's in a flex box mock-ups we have (see #4295). I know there was some discourse on the style and consistency throughout, so happy to change if necessary. 

![Screenshot from 2024-02-12 13-21-37](https://github.com/containers/podman-desktop/assets/87311656/a1d4c978-2953-4c94-8d0a-bd6a0f6bc60d)

**Comment by benoitf** [comment]
hello, maybe it's just me but I'm not sure I would find it more convenient. I'm not sure if these buttons are for the upper form as we have several different background colors
It's like if the buttons are not related to something

**Comment by deboer-tim** [comment]
Ditto for me, I like the fixed position but it loses some of the connection with the form.

**Comment by cdrage** [comment]
I do agree as well that they seem disconnected.

Could there perhaps be two designs going forward that would help? One that's "full page" with the fixed buttons.

The other always having two buttons (one with cancel, the other with the executing button?). So it's more consistent throughout PD.

For example, switching from 1 button:
![Screenshot 2024-02-12 at 8 46 10 AM](https://github.com/containers/podman-desktop/assets/6422176/89cc6c86-36f8-4b8a-a0aa-218e20d3bd2f)

To two buttons (cancel goes back):
![Screenshot 2024-02-12 at 8 46 03 AM](https://github.com/containers/podman-desktop/assets/6422176/18bf1845-3f62-4bd3-85a1-0cd5e84b4b71)

**Comment by cdrage** [comment]
![Screenshot 2024-02-12 at 8 48 10 AM](https://github.com/containers/podman-desktop/assets/6422176/a2bd4a39-2415-4390-b348-04a4190edb19)

The background for these forms look different vs resources too. So we have I think 3 different backgrounds.

So maybe updating each form (pull, build, create container), to similar layout as resources?

**Comment by ekidneyrh** [comment]
Hmm I want to make sure I'm understanding right.

In terms of the colours, I'm following the [colour doc](https://design.penpot.app/#/workspace/82e31d90-3829-8139-8002-cb67c32e72ca/8ec95363-4e2c-80a4-8002-cb67c874d0be?page-id=b2d10cc9-d8f7-8046-8003-db40572d0468) that's in penpot for the dark mode at the moment as it is being made.

For the background colours I have been going with charcoal-700 (#222222) for non-setting's pages and charcoal-800 (#18181b) for settings pages that I saw mentioned [here](https://design.penpot.app/#/workspace/82e31d90-3829-8139-8002-cb67c32e72ca/8ec95363-4e2c-80a4-8002-cb67c874d0be?page-id=42e69059-b409-807e-8002-d0b98cda8446):

![Screenshot from 2024-02-12 14-04-01](https://github.com/containers/podman-desktop/assets/87311656/a38a85e0-7818-454e-b5c6-45dcc82d5433)

Have the background colours been updated since this was created? I'm trying to stick to what's definite, so things are consistent. I know once the colour theming is in place, it'll help sort out inconsistencies that are present at the moment. 

The example I gave was a non-settings page, so I used charcoal-700.

For the buttons, I want to make sure I understand properly - so this ticket is for the full page with fixed buttons. Ones that are always present on the bottom right of the page that float on top of the content like the onboarding screens:
![image](https://github.com/containers/podman-desktop/assets/87311656/0a030595-1354-46e4-828a-a3510d2a8a53)

The other/alternative design being just the normal two button design that at the end of a page - like this for screen mock-up:
![Screenshot from 2024-02-12 14-21-36](https://github.com/containers/podman-desktop/assets/87311656/1d5cbf5b-2e7a-448b-a198-7b1c90e9dea4)

And are we looking to replace the one button stretched with the normal two button design?

**Comment by cdrage** [comment]
> Hmm I want to make sure I'm understanding right.
> 
> In terms of the colours, I'm following the [colour doc](https://design.penpot.app/#/workspace/82e31d90-3829-8139-8002-cb67c32e72ca/8ec95363-4e2c-80a4-8002-cb67c874d0be?page-id=b2d10cc9-d8f7-8046-8003-db40572d0468) that's in penpot for the dark mode at the moment as it is being made.
> 
> For the background colours I have been going with charcoal-700 (#222222) for non-setting's pages and charcoal-800 (#18181b) for settings pages that I saw mentioned [here](https://design.penpot.app/#/workspace/82e31d90-3829-8139-8002-cb67c32e72ca/8ec95363-4e2c-80a4-8002-cb67c874d0be?page-id=42e69059-b409-807e-8002-d0b98cda8446):
> 
> ![Screenshot from 2024-02-12 14-04-01](https://private-user-images.githubusercontent.com/87311656/304112617-a38a85e0-7818-454e-b5c6-45dcc82d5433.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc3NTE4OTEsIm5iZiI6MTcwNzc1MTU5MSwicGF0aCI6Ii84NzMxMTY1Ni8zMDQxMTI2MTctYTM4YTg1ZTAtNzgxOC00NTRlLWI1YzYtNDVkY2M4MmQ1NDMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjEyVDE1MjYzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVmZDlkZGU4NTY3ZGJkOTlhOTVkNTQ1MzUyOTYxMWRhNzc0MGJjYjc4MmRmZTA4NmJiYmJlNTk3Y2Q3NzRkNzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xQ47gT_B7ZJ2Z1yU7qZeX-yj4ZVO15csTZACurMzbTE)
> 
> Have the background colours been updated since this was created? I'm trying to stick to what's definite, so things are consistent. I know once the colour theming is in place, it'll help sort out inconsistencies that are present at the moment.
> 
> The example I gave was a non-settings page, so I used charcoal-700.
> 
> For the buttons, I want to make sure I understand properly - so this ticket is for the full page with fixed buttons. Ones that are always present on the bottom right of the page that float on top of the content like the onboarding screens: ![image](https://private-user-images.githubusercontent.com/87311656/304119291-0a030595-1354-46e4-828a-a3510d2a8a53.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc3NTE4OTEsIm5iZiI6MTcwNzc1MTU5MSwicGF0aCI6Ii84NzMxMTY1Ni8zMDQxMTkyOTEtMGEwMzA1OTUtMTM1NC00NmU0LTgyOGEtYTM1MTBkMmE4YTUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjEyVDE1MjYzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU2ZDEyMDk4NTUyNWFkNTAwNGIzNWQ1ZTQ4YzY1YzUwNWU3NDAwNWVmNzBkNmRjNTI3YzFmNTc5Y2QyMzU1YjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5aFk9M52Z6QHxKA75VS4u92yg5hjHa_S8SYF-pDqZGE)
> 
> The other/alternative design being just the normal two button design that at the end of a page - like this for screen mock-up: ![Screenshot from 2024-02-12 14-21-36](https://private-user-images.githubusercontent.com/87311656/304118062-1d5cbf5b-2e7a-448b-a198-7b1c90e9dea4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc3NTE4OTEsIm5iZiI6MTcwNzc1MTU5MSwicGF0aCI6Ii84NzMxMTY1Ni8zMDQxMTgwNjItMWQ1Y2JmNWItMmU3YS00NDhiLWExOTgtN2IxYzkwZTlkZWE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjEyVDE1MjYzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMzYTNkZGI1YzdiMWZjNTAwMjE2NjU4NWEwNjQzZjk4ZDQ4MTY1MTZmZDlkODQxYWM0NjNiZjU5NzkxYWMyMTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.S7XICzK3irhmwqdadC8e_gNZN7_0LOc2xtQF6qNCe6g)
> 
> And are we looking to replace the one button stretched with the normal two button design?

At this point I think that's where the benefit would be and switching to a two-button design that matches all the other screens / always have buttons bottom right.

It would also be beneficial having that "Close" button that'll take you back to the previous screen / easy back and forth, rather than trying to find the URL (usually "Pods" on the top left).

Thank you for clarifying about the background colours, now that I examined it more closely they are being followed correctly! You're right that they are only two background colours. Probably another reason why i'm looking forward to light mode was that I was finding it difficult distinguishing between the two haha.

I think we should get a few more people's opinions on the two-button design rather than the large-stretched out button.

WDYT @deboer-tim @benoitf @feloy @lstocchi ?

**Comment by benoitf** [comment]
my 2 cents is that we should focus the work on the light theme
This issue has been created 3 days ago, we've long standing missing design that are not yet there, so probably this can be tackled later. Not saying it's not important but focus on priorities

**Comment by ekidneyrh** [comment]
do we think this is still relevant?

**Comment by deboer-tim** [comment]
We are still inconsistent; at some point we should confirm the design and apply it to the places that don't match.

**Comment by ekidneyrh** [comment]
I think we could switch to having the two bottom right buttons throughout the app - would we keep the 'X' on the top right if we did this? would it make sense to have two ways to close the form? 

Also we have screens like this:

![image](https://github.com/user-attachments/assets/6e394585-5c65-4a11-85fa-ea761a5bf5d8)

where there a box underneath the action button - would it make sense to move this above the buttons so we continue with have that action and close button on the bottom right of the page?

**Comment by deboer-tim** [comment]
> I think we could switch to having the two bottom right buttons throughout the app - would we keep the 'X' on the top right if we did this? would it make sense to have two ways to close the form?

I think this is your call :). I find the couple of screens where we have both ways inconsistent today though, so I'd prefer if we went with all forms having a cancel button or all forms only have an X, never both.

> Also we have screens like this:
>
> where there a box underneath the action button - would it make sense to move this above the buttons so we continue with have that action and close button on the bottom right of the page?

I dislike this pattern either way, since both positions have usability issues. Ditto for the current pages where the console suddenly appears when you click the action. I'd rather the form and 'running task' were separate pages, e.g. once you click the action the form collapses or goes away and you get a different UI with the task name or summary, progress, console, and would be more consistent across different tasks.

**Comment by ekidneyrh** [comment]
We can go with removing the X button then in favour of displaying a cancel / close button along side the action - it removes the double X that comes up on forms :D 

> I dislike this pattern either way, since both positions have usability issues. Ditto for the current pages where the console suddenly appears when you click the action. I'd rather the form and 'running task' were separate pages, e.g. once you click the action the form collapses or goes away and you get a different UI with the task name or summary, progress, console, and would be more consistent across different tasks.

For now I think we should move both buttons underneath the console box and open another ticket for changing how we display the running tasks actions like that on form pages
